core/src/main/java/org/sonarsource/sonarlint/core/container/connected/update/PluginReferencesDownloader.java

```diff
@@ -82,9 +81,11 @@ private static boolean checkVersion(@Nullable String version, @Nullable String m
     return true;
   }
 
-  public PluginReferences fetchPluginsTo(Path dest, List<SonarAnalyzer> analyzers) {
+  public PluginReferences fetchPluginsTo(Path dest, List<SonarAnalyzer> analyzers, ProgressWrapper progress) {
     PluginReferences refs = fetchPlugins(analyzers);
+    int i = 0;
     for (PluginReference ref : refs.getReferenceList()) {
+      progress.setProgressAndCheckCancel("Loading plugin " + ref.getKey(), i / (float) refs.getReferenceList().size());
```

Comment:
I would extract the constant expression `(float) refs.getReferenceList().size()` to a variable before the loop.